internal/grpcserver/storage/elastic.go

```diff
@@ -0,0 +1,213 @@
+package storage
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"github.com/astanishevskyi/grpc-server/internal/grpcserver/models"
+	elastic "github.com/elastic/go-elasticsearch/v7"
+	"github.com/elastic/go-elasticsearch/v7/esapi"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+type ElasticStorage struct {
+	client *elastic.Client
+	mu     sync.Mutex
+	lastID uint32
+}
+
+func NewElasticStorage() *ElasticStorage {
+	user := os.Getenv("ELASTIC_USER")
+	pass := os.Getenv("ELASTIC_PASS")
+	addr := os.Getenv("ELASTIC_ADDR")
+
+	client, err := elastic.NewClient(elastic.Config{
+		Addresses: []string{addr},
+		Username:  user,
+		Password:  pass,
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+	if _, err := client.Ping(); err != nil {
```

Comment:
every service should have a health check server launched on the separate port to notify orchestration system when smth went wrong. The `ping` method should be a part of `Storage` class just to have an ability to check whether the storage connection is alive. As a next step you might be willing to add some reconnection attempt logic